src/mocompiler.js

```diff
@@ -183,52 +199,53 @@ Compiler.prototype._calculateSize = function (list) {
 /**
  * Generates the binary MO object from the translation list
  *
- * @param {import('./index.d.ts').GetTextTranslations} list translation list
- * @param {Object} size Byte size information
- * @return {Buffer} Compiled MO object
+ * @param {import('./types.js').GetTextTranslation[]} list translation list
+ *  @param {{ msgid: number; msgstr: number; total: number }} size Byte size information
+ *  @return {Buffer} Compiled MO object
  */
 Compiler.prototype._build = function (list, size) {
+  /** @type {Buffer} returnBuffer */
   const returnBuffer = Buffer.alloc(size.total);
   let curPosition = 0;
   let i;
   let len;
 
   // magic
-  returnBuffer[this._writeFunc](this.MAGIC, 0);
+  returnBuffer.writeUInt32LE(this.MAGIC, 0);
 
   // revision
-  returnBuffer[this._writeFunc](0, 4);
+  returnBuffer.writeUInt32LE(0, 4);
 
   // string count
-  returnBuffer[this._writeFunc](list.length, 8);
+  returnBuffer.writeUInt32LE(list.length, 8);
 
   // original string table offset
-  returnBuffer[this._writeFunc](28, 12);
+  returnBuffer.writeUInt32LE(28, 12);
 
   // translation string table offset
-  returnBuffer[this._writeFunc](28 + (4 + 4) * list.length, 16);
+  returnBuffer.writeUInt32LE(28 + (4 + 4) * list.length, 16);
 
   // hash table size
-  returnBuffer[this._writeFunc](0, 20);
+  returnBuffer.writeUInt32LE(0, 20);
 
   // hash table offset
-  returnBuffer[this._writeFunc](28 + (4 + 4) * list.length * 2, 24);
+  returnBuffer.writeUInt32LE(28 + (4 + 4) * list.length * 2, 24);
```

Comment:
I don't believe `[this._writeFunc]` should be removed, similar to my comment in `moparser.js`

Reply:
As per the conversation we had privately I remove changes temporarily as that may introduce problems to do so in a dedicated pr